### PR TITLE
RoCC PTW refactoring

### PR DESCRIPTION
**Pull request by yunsup**
@aswaterman @zhemao cleaning up the RoCC PTW ports.

### CODE REVIEW

**Comment by colinschmidt** [comment]
looks good to me

**Comment by ccelio** [comment]
Glad to see you still have time to write code ;)

File: src/main/scala/tile.scala [hunk 1]
```diff
@@ -88,12 +90,13 @@ class RocketTile(resetSignal: Bool = null)(implicit p: Parameters) extends Tile(
...
-      ptw.io.requestor(2 + 3 * i) <> rocc.io.iptw
-      ptw.io.requestor(3 + 3 * i) <> rocc.io.dptw
-      ptw.io.requestor(4 + 3 * i) <> rocc.io.pptw
       rocc
     }
 
+    if (p(RoccNPTWPorts) > 0) {
+      ptw.io.requestor.drop(2) <> roccs.flatMap(_.io.ptw)
+    }
+
```
Comment (aswaterman) [inline src/main/scala/tile.scala]: doesn't this worth without the if-statement?

Comment (yunsup) [inline src/main/scala/tile.scala]: Again, I'm trying to be consistent with the other parameterization going on in other places: if (nFPUPorts > 0), if (p(RoccNCSRs) > 0).  One could argue that it's better documentation: connect these when there are the PTW ports are requested.

Comment (aswaterman) [inline src/main/scala/tile.scala]: OTOH, you would never write C code like

    if (n > 0)
      for (i = 0; i < n; i++)

but you make the call.

Comment (yunsup) [inline src/main/scala/tile.scala]: Albert said he was more inclined without the if statement, so I'm going ahead and submitting a new PR.  Although I'm keeping Rocc as is, since changing Rocc to RoCC should be another PR.
